### ISSUE TITLE
perlcc's -o option and e_perlcc.t dont agree on how to name a binary on Win32

### ISSUE THREAD

**Issue by bulk88** [rurban/perl-compiler#371]
This is a bug in cperl on Win32 in 'e_perlcc.t'.

e_perlcc.t has a line like

```
is(`$perlcc -S -o pcc -r $f $devnull`, "ok", "-S -o -r file");
```

The -o option is hard coded to say just "pcc", not "pcc.exe". On Win32 with Visual C (I didn't test Mingw GCC), the VC linker produces a file called "pcc". The "pcc" is a valid, PE headered, Win32 executable, but the later in e_perlcc.t there is

```
ok(-e $a, "keep executable");
cleanup;
```
and $a is "pcc.exe" not "pcc", so the test fails.

The docs for perlcc's -o option say 

```
=item -o I<output file name>

Specifies the file name for the final compiled executable.

Without given output file name we use the base of the input file,
or with C<-e> F<a.out> resp. F<a.exe> and a randomized intermediate
C filename.
If the input file is an absolute path on a non-windows system use
the basename.
```

So which file is responsible for making sure the VC linker produces a file that ends in .exe? You can argue it is the user's fault for not passing ".exe" to the -o option, so e_perlcc.t needs to be fixed, or you can argue that it is perlcc's fault, since an extension-less PE executable on Win32 is completely abnormal and there is no valid use for making/linking an extension-less PE file, this isn't Unix.

**Comment by rurban** [comment]
I would say, `-o pcc` should produce `pcc.exe`, so perlcc needs to be fixed. I just changed that code recently and introduced this bug.

**Comment by rurban** [comment]
Fixed with d10671902b13201e73f941e4493275b1379a4940
